test/Microsoft.OData.ModelBuilder.Tests/Vocabularies/Capabilities/DeleteRestrictionsTests.cs

```diff
@@ -0,0 +1,128 @@
+        private const string CustomerReadWriteAllScope = "Customers.ReadWrite.All";
+
+        [Fact]
+        public void DeleteRestrictionsChainsToSameInstance()
+        {
+            var modelBuilder = new ODataModelBuilder().Add_Customers_EntitySet();
+            var entitySetConfiguration = modelBuilder.EntitySet<Customer>("Customers");
+
+            var builder1 = entitySetConfiguration.HasDeleteRestrictions();
+            var builder2 = builder1.IsDeletable(true);
+
+            Assert.Same(builder1, builder2);
+        }
+
+        [Fact]
+        public void AnnotationsAreAddedToModelOnBuild()
+        {
+            var modelBuilder = new ODataModelBuilder().Add_Customer_EntityType().Add_Customers_EntitySet();
+            var entitySetConfiguration = modelBuilder.EntitySet<Customer>("Customers");
+
+            var deleteRestrictionsBuilder = entitySetConfiguration
+                .HasDeleteRestrictions()
+                .IsDeletable(false)
+                .HasDescription(Description);
+
+            var model = modelBuilder.GetServiceModel();
+            var container = model.SchemaElements.OfType<IEdmEntityContainer>().SingleOrDefault();
+            var customers = container.EntitySets().SingleOrDefault(e => e.Name == "Customers");
+
+            var term = model.FindTerm("Org.OData.Capabilities.V1.DeleteRestrictions");
+            var annotation = customers.VocabularyAnnotations(model).FirstOrDefault(a => a.Term == term);
+            Assert.NotNull(annotation);
+
+            var annotationValue = annotation.Value as EdmRecordExpression;
+            Assert.NotNull(annotationValue);
+
+            var descriptionProperty = annotationValue.FindProperty("Description");
+            Assert.NotNull(descriptionProperty);
+
+            var stringValue = descriptionProperty.Value as IEdmStringValue;
+            Assert.NotNull(stringValue);
+
+            Assert.Equal(Description, stringValue.Value);
+        }
+
+        [Fact]
+        public void PermissionsCanBeAddedToEntitySet()
+        {
+            var modelBuilder = new ODataModelBuilder().Add_Customer_EntityType().Add_Customers_EntitySet();
+            var entitySetConfiguration = modelBuilder.EntitySet<Customer>("Customers");
+
+            var readWritePermission = new PermissionType { SchemeName = SchemeName };
+            readWritePermission.Scopes.AddRange(new ScopeType[]
+            {
+                new ScopeType
+                {
+                    Scope = CustomerReadWriteAllScope,
+                    RestrictedProperties = "*"
+                }
+            });
+
+            var deleteRestrictionsBuilder = entitySetConfiguration
+                .HasDeleteRestrictions()
+                .IsDeletable(false)
+                .HasDescription(Description)
+                .HasPermissions(new[]
+                {
+                    readWritePermission
+                })
+                .HasLongDescription(LongDescription);
```

Comment:
I think it should be easier to add basic permissions without having to break the "fluent" flow.
e.g.
```c#
entitySetConfiguration
.HasDeleteRestrictions()
.IsDeletable(false)
.HasDescription(description)
.HasPermission(p => p.HasScheme("Personal").HasScope("Calendar.Write", "+Name,+Date"))
.HasPermission(p => p.HasScheme("Delegated").HasScope("Calendar.Read", "-Name"))
.HasLongDescription(longDescription)
```
So basically, instead of `HasPermissions`, you add one permission record a time with `HasPermission` with takes a lambda that has a `PermissionBuilder` as parameter which can be used to build a permission record inline.

Reply:
Another question, if someone wanted to add permissions for different actions, would those require two separate "fluent" statements? i.e.

```c#
entitySetConfiguration.HasDeleteRestrictions().IsDeletable()...;

entitySetConfiguration.HasReadRestrictions()...;
```

Reply:
For the first comment, Yes!
For the second comment/question, yes, since the two builders would be having their own set of HasXs that they contain...

Reply:
I'm not sure I fully support the direction being proposed here wrt the `HasPermission` args. At this point you are also adding the fluent syntax to the `PermissionType` itself (`p.HasScheme(...).HasScope(...)`) and not just the `entitySetConfiguration`.